### ISSUE TITLE
`pipe` is not a reliable way to detect node-streams

### ISSUE THREAD

**Issue by jamestalmage** [tunnckoCore/is-node-stream#2]
The existence of a `pipe` method on Writable is not documented.
It does exist (throws an error), but relying on undocumented internals seems like a bad way to go.

I think it is better to do it like this:

``` js
isReadable: (stream) =>
  isEmitter(stream) && 
  isFunction(stream.pipe) && (
    isFunction(stream.read) || 
    (isFunction(stream.pause) && isFunction(stream.resume))
  ),
isWritable: (stream) =>
  isEmitter(stream) && 
  isFunction(stream.write) &&
  isFunction(stream.end)
```

**Comment by tunnckoCore** [comment]
Hm. Good point, but this is core documentation problem, imo. Didn't noticed it, lol.
You can open issue to docs site and we can wait with that move. Because this same problem can be seen in https://github.com/sindresorhus/is-stream and if you want you can ping sindre here or open same issue on his package, cuz i'm blocked.

**Comment by jamestalmage** [comment]
> this is core documentation problem, imo.

Not necessarily. Why document a method no one is supposed to use? You can't pipe a writable anywhere, it wouldn't make sense. (Which is pretty much what the comment in the source and the thrown error say). It would be perfectly reasonable for them to remove it completely at some point down the road (at which point people start getting `TypeError: undefined is not a function`). What would the documentation say?

I've seen Sindre's, and yeah - it has the same issue. I think I am going with his. (With 19k downloads a day somebody is going to notice real fast if it stops working).

**Comment by tunnckoCore** [comment]
Hah, downloads. I'm all about strictness (even call me "strict guy").
So you can PR if you want, you're welcome.

**Comment by tunnckoCore** [comment]
> Not necessarily. Why document a method no one is supposed to use? You can't pipe a writable anywhere, it wouldn't make sense.

Basically, yes, but it's not deprecated in no one of the stable versions from 0.10 to 4, so users can rely on it, for things like these checks, even if it not make sense really, as you said.
